### PR TITLE
Feature/jr2 1224 cancel loopback

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/events/JobAdvertisementCancelledEvent.java [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 /**
  * JobAdvertisement has been cancelled be the user
  */
 public class JobAdvertisementCancelledEvent extends JobAdvertisementEvent {
 
-    public JobAdvertisementCancelledEvent(JobAdvertisement jobAdvertisement) {
+    public JobAdvertisementCancelledEvent(JobAdvertisement jobAdvertisement, SourceSystem cancelledBy) {
         super(JobAdvertisementEvents.JOB_ADVERTISEMENT_CANCELLED, jobAdvertisement);
+        additionalAttributes.put("cancelledBy", cancelledBy);
```
Comment (pavax) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/events/JobAdvertisementCancelledEvent.java]: Precondition not null for cancelledBy

File: infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListener.java [hunk 2]
```diff
@@ -26,7 +27,17 @@ protected void handle(JobAdvertisementInspectingEvent event) {
 
     @EventListener
     protected void handle(JobAdvertisementCancelledEvent event) {
-        LOG.debug("EVENT caught for AVAM: JOB_ADVERTISEMENT_CANCELLED for JobAdvertisementId: '{}'", event.getAggregateId().getValue());
-        this.avamTaskRepository.save(new AvamTask(new AvamTaskId(), event.getAggregateId(), AvamTaskType.DEREGISTER));
+        if (shouldBeSend(event)) {
```
Comment (pavax) [inline infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListener.java]: code style -> invert if

File: infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamService.java [hunk 3]
```diff
@@ -96,6 +97,6 @@ public void handleCancelAction(CancellationDto cancellationDto) {
             jobAdvertisementDto = jobAdvertisementApplicationService.findByStellennummerAvam(cancellationDto.getStellennummerAvam());
         }
         Condition.notNull(jobAdvertisementDto, "Couldn't find the jobAdvertisement for stellennummerEgov %s nor stellennummerAvam %s", cancellationDto.getStellennummerEgov(), cancellationDto.getStellennummerAvam());
-        jobAdvertisementApplicationService.cancel(new JobAdvertisementId(jobAdvertisementDto.getId()), cancellationDto.getDate(), cancellationDto.getCode(), null);
+        jobAdvertisementApplicationService.cancel(new JobAdvertisementId(jobAdvertisementDto.getId()), cancellationDto.getDate(), cancellationDto.getCode(), SourceSystem.RAV, null);
```
Comment (pavax) [inline infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamService.java]: code formatting -> use new lines

File: infrastructure-messagebroker/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        JobAdvertisementCancelledEvent event = new JobAdvertisementCancelledEvent(JOBADVERTISEMENT, SourceSystem.RAV);
+
+        // WHEN
+        sut.handle(event);
+
+        // THEN
+        verify(avamTaskRepository, never()).save(any());
+    }
+
+}
```
Comment (pavax) [inline infrastructure-messagebroker/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListenerTest.java]: new line after eof

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/events/JobAdvertisementCancelledEvent.java (hunk 1 above)
Comment (cjpsycos) [reply domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/events/JobAdvertisementCancelledEvent.java]: done

File: infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListener.java (hunk 2 above)
Comment (cjpsycos) [reply infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListener.java]: done

File: infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamService.java (hunk 3 above)
Comment (cjpsycos) [reply infrastructure-messagebroker/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamService.java]: done

File: infrastructure-messagebroker/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListenerTest.java (hunk 4 above)
Comment (cjpsycos) [reply infrastructure-messagebroker/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/messagebroker/avam/AvamRegistrationEventListenerTest.java]: done

**Pull request merged**